src/transformers/trainer.py

```diff
@@ -239,6 +240,10 @@ def __init__(
         self.hp_name = None
         self.deepspeed = None
 
+        # memory metrics - must set up as early as possible
+        self.mem = TrainerMemoryTracker(self.args.skip_memory_metrics)
```

Comment:
A bit too short IMO. Also does this need to be a public attribute? Maybe `self._memory_tracker` would be better.

Reply:
done